logrusx/hooks.go

```diff
@@ -0,0 +1,41 @@
+package logrusx
+
+import "github.com/sirupsen/logrus"
+
+type requestIdHook struct {
+	rIdCtxKey   interface{}
+	rIdFieldKey string
+}
+
+var _ logrus.Hook = (*requestIdHook)(nil)
+
+func NewRequestIdHook(requestIdContextKey interface{}, requestIdFieldKey string) *requestIdHook {
+	return &requestIdHook{
+		rIdCtxKey:   requestIdContextKey,
+		rIdFieldKey: requestIdFieldKey,
+	}
+}
+
+func (irh *requestIdHook) Levels() []logrus.Level {
+	return logrus.AllLevels
+}
+
+func (rih *requestIdHook) Fire(entry *logrus.Entry) (outErr error) {
+	defer func() {
+		// Nullify panic to prevent having this hook break a request
+		if rec := recover(); rec != nil {
+			outErr = nil
+		}
```

Comment:
Seems irrelevant here, I'd just recover if the goal is to be a guardrail
```suggestion
		recover()
```